src/main/java/org/threadly/util/debug/Profiler.java

```diff
@@ -231,6 +311,30 @@ public void run() {
             throw ExceptionUtils.makeRuntime(e.getCause());
           }
         }
+        // start or schedule to handle run time limit
+        if (sampleDurationInMillis > 0) {
+          if (executor instanceof SimpleSchedulerInterface) {
```

Comment:
I think I agree its probably just not worth taking in an executor and just create the extra thread.  If someone is running the profiler they already know there is extra over head.